server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -568,34 +570,57 @@ private ExtensionsSettings readFromExtensionsYml(Path filePath) throws IOExcepti
             List<HashMap<String, ?>> unreadExtensions = new ArrayList<>((Collection<HashMap<String, ?>>) obj.get("extensions"));
             List<Extension> readExtensions = new ArrayList<Extension>();
             for (HashMap<String, ?> extensionMap : unreadExtensions) {
-                // Parse extension dependencies
-                List<ExtensionDependency> extensionDependencyList = new ArrayList<ExtensionDependency>();
-                if (extensionMap.get("dependencies") != null) {
-                    List<HashMap<String, ?>> extensionDependencies = new ArrayList<>(
-                        (Collection<HashMap<String, ?>>) extensionMap.get("dependencies")
-                    );
-                    for (HashMap<String, ?> dependency : extensionDependencies) {
-                        extensionDependencyList.add(
-                            new ExtensionDependency(
-                                dependency.get("uniqueId").toString(),
-                                Version.fromString(dependency.get("version").toString())
-                            )
+                try {
+                    // iterating the map to see whether any required field is missing
+                    String[] requiredFields = {
+                        "name",
+                        "uniqueId",
+                        "hostAddress",
+                        "port",
+                        "version",
+                        "opensearchVersion",
+                        "minimumCompatibleVersion" };
+                    for (String requireField : requiredFields) {
+                        if (!extensionMap.containsKey(requireField)) {
+                            throw new ParsingException(
+                                new XContentLocation(584, 10),
+                                "Extension : " + extensionMap.get("name") + " is missing this required field : " + requireField
```

Comment:
if "name" is missing, you won't be able to retrieve the name here for the error message!

Suggestion: create a list or set of missing fields, and as you iterate, add any missing elements to the set at this point.  Then check if it's empty, if so, great. If not, throw an exception and list the missing fields in the message.

Reply:
Hi

Yes i will mention the missing fields in log but since yml file can have more than one extension, i thought it would be better for the user to understand which extension had missing fields and did load but yeah since any field could be missing, we cannot mention its name or any unique Id

Thanks

Reply:
changed as per the review

Reply:
Good point, I guess if the name exists you could use it, otherwise this would say null if the name is missing.